src/main/java/mate/academy/dao/jdbc/CarDaoJdbc.java

```diff
@@ -0,0 +1,204 @@
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model_car = ?, "
+                + "manufacturer_id = ? WHERE id_car = ? "
+                + "AND delete_car = FALSE";
+        deleteAllDriversFromCar(car);
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+            for (Driver driver : car.getDrivers()) {
+                addDriverToCar(driver, car);
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Could update car " + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET delete_car = TRUE "
+                + "WHERE id_car = ? ";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            int delete = statement.executeUpdate();
+            return delete > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Could not delete car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT cars.id_car, cars.model_car, cars.manufacturer_id, "
+                + "manufacturer.name_manufacturer, manufacturer.country_manufacturer "
+                + "FROM cars "
+                + "JOIN cars_drivers ON cars.id_car = cars_drivers.id_car "
+                + "JOIN manufacturer ON cars.manufacturer_id = manufacturer.id_manufacturer "
+                + "WHERE cars_drivers.id_driver = ? "
+                + "AND delete_driver = FALSE "
+                + "AND delete_car = FALSE";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(createCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Could not get all Cars by Driver id " + driverId, e);
+        }
+        return cars;
+    }
+
+    private void addDriverToCar(Driver driver, Car car) {
+        String query = "INSERT INTO cars_drivers (id_car, id_driver) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, car.getId());
```

Comment:
```suggestion
    private void addDriverToCar(Car car) {
        String query = "INSERT INTO cars_drivers (id_car, id_driver) VALUES (?, ?);";
        try (Connection connection = ConnectionUtil.getConnection();
                PreparedStatement statement = connection.prepareStatement(query)) {
                
            // for each driver: 
            statement.setLong(1, car.getId());
            ...
```
let's not open connections in a loop, we can execute statement multiple times in one connection instead